cypress/e2e/favouriting.cy.ts

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from 'uuid'
+
+describe('Model favouriting', () => {
+  const modelName = `Test model ${uuidv4()}`
+
+  before(() => {
+    cy.log('Navigate to Upload page and json tab')
+    cy.visit('/upload')
+    cy.get('[data-test=uploadJsonTab]').click({ force: true })
+
+    cy.log('Selecting schema and inputting metadata')
+    cy.get('[data-test=selectSchemaInput]').trigger('mousedown', { force: true, button: 0 })
+    cy.fixture('schema_names.json').then((schemaNames) => {
+      cy.get(`[role=option]:contains(${schemaNames.model})`).click()
+    })
+    cy.fixture('minimal_metadata.json').then((metadata) => {
+      const updatedMetadata = { ...metadata }
+      updatedMetadata.highLevelDetails.name = modelName
+      updatedMetadata.buildOptions.uploadType = 'Model card only'
+      cy.get('[data-test=metadataTextarea]')
+        .clear()
+        .type(JSON.stringify(metadata), { parseSpecialCharSequences: false, delay: 0 })
+    })
+
+    cy.log('Submitting model')
+    cy.get('[data-test=warningCheckbox]').click()
+    cy.get('[data-test=submitButton]').click()
+    cy.url({ timeout: 10000 }).as('modelUrl').should('contain', '/model/')
```

Comment:
I've done the following in my tests to make these url checks a bit stronger:
```
      import convertNameToUrlFormat from '../utils/convertNameToUrlFormat'

      cy.url({ timeout: 10000 })
        .as('modelUrl')
        .should('contain', `/model/${convertNameToUrlFormat(modelMetadata.highLevelDetails.name)}`)
        
```

In this case, if you wanted to do this, you'd have to move lines 25-28 inside your `.then()` callback on line 16 to have access to the fixture.